### PR TITLE
Adds markdownlint

### PR DESCRIPTION
**Pull request by sanjeetsuhag**
This PR adds support for linting our Markdown documentation through `markdown-cli`.

Open questions:
- Should this be part of CI?

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @sanjeetsuhag!
* :heavy_check_mark: Signed CLA found.
* :grey_question: [CHANGES.md](https://github.com/CesiumGS/cesium/blob/markdown-lint/CHANGES.md) was not updated.
    * If this change updates the public API in any way, please add a bullet point to `CHANGES.md`.
* :grey_question: Unit tests were not updated.
    * Make sure you've [updated tests](https://github.com/CesiumGS/cesium/tree/main/Documentation/Contributors/TestingGuide) to reflect your changes, added tests for any new code, and ran the code coverage tool.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.

**Review by ggetz — COMMENTED** [review]
Awesome, thanks @sanjeetsuhag! This works great and will come in handy as we update our guides. It also runs fast, so I think it makes sense to add it to CI. Just make sure it fails the task if the linter catches a warning or error, or else it won't fail the build.

File: package.json [hunk 1]
```diff
@@ -133,6 +134,7 @@
     "generateDocumentation-watch": "gulp generateDocumentation-watch",
     "eslint": "eslint \"./**/*.js\" \"./**/*.cjs\" \"./**/*.html\" --cache --quiet",
     "makeZipFile": "gulp makeZipFile",
+    "markdownlint": "markdownlint \"Documentation/**/*.md\"",
```
[the remark this answers is not in the document]
Comment (sanjeetsuhag) [reply package.json]: They run on root level `.md` files too, except `README.md`, `CHANGES.md` and `LICENSE.md`. For these 3 files, I think we want to remain flexible.

**Comment by ggetz** [comment]
Awesome, thanks @sanjeetsuhag!

**Pull request merged**